opengrok-indexer/src/main/java/org/opengrok/indexer/util/HeadHandler.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2020 Oracle and/or its affiliates. All rights reserved.
+ */
+
+package org.opengrok.indexer.util;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The purpose of this class is to provide StreamHandler that limits the output
+ * to specified number of lines. Compared to {@code SpoolHandler} it consumes
+ * limited amount of heap.
+ */
+public class HeadHandler implements Executor.StreamHandler {
+    private int numlines;
+
+    private final List<String> lines = new ArrayList<>();
+
+    // TODO add encoding
+    public HeadHandler(int numlines) {
+        this.numlines = numlines;
+    }
+
+    public List<String> getLines() {
+        return lines;
```

Comment:
I recommend to avoid breaking encapsulation (though it's rampant in OpenGrok). E.g. rather than returning a private list, how about:

```
public int count() {
...
}

public String get(int index) {
...
}
```
```

Reply:
I must admit OpenGrok code base has been primary source of my Java programming influence :-)